purchase_request_department/models/purchase_request.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2017-2020 Forgeflow S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl-3.0).
+from odoo import api, fields, models
+
+
+class PurchaseRequest(models.Model):
+    _inherit = "purchase.request"
+
+    def _get_my_department(self):
+        employees = self.env.user.employee_ids
+        return (
+            employees[0].department_id
+            if employees
+            else self.env["hr.department"] or False
+        )
+
+    department_id = fields.Many2one(
+        "hr.department", "Department", default=_get_my_department
+    )
+
+    @api.onchange("requested_by")
+    def onchange_requested_by(self):
+        employees = self.requested_by.employee_ids
+        self.department_id = (
+            employees[0].department_id
```

Comment:
Non-blocking, but I would prefer to ensure that there is no error even len(employees) = 0
```suggestion
            employees[:1].department_id
```